src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,136 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (first == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (getNode(index) == first) {
+            newNode.next = getNode(index);
+            first = newNode;
+        } else {
+            newNode.prev = getNode(index).prev;
+            newNode.next = getNode(index).next;
+            getNode(index).prev.next = newNode;
+        }
+        getNode(index).prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T changedValue = get(index);
+        getNode(index).value = value;
+        return changedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removedValue = get(index);
+        unlink(getNode(index));
+        size--;
+        return removedValue;
```

Comment:
```suggestion
        return unlink(getNode(index));
```

move `size--` to `unlink`

Reply:
done